hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -52,29 +52,18 @@
  */
 public class DataSourceUtils {
 
-  /**
-   * Obtain value of the provided nullable field as string, denoted by dot notation. e.g: a.b.c
-   */
-  public static String getNullableNestedFieldValAsString(GenericRecord record, String fieldName) {
-    try {
-      return getNestedFieldValAsString(record, fieldName);
-    } catch (HoodieException e) {
-      return null;
-    }
-  }
-
   /**
    * Obtain value of the provided field as string, denoted by dot notation. e.g: a.b.c
    */
-  public static String getNestedFieldValAsString(GenericRecord record, String fieldName) {
-    Object obj = getNestedFieldVal(record, fieldName);
-    return obj.toString();
+  public static String getNestedFieldValAsString(GenericRecord record, String fieldName, boolean returnNullIfNotFound) {
+    Object obj = getNestedFieldVal(record, fieldName, returnNullIfNotFound);
+    return (obj == null) ? null : obj.toString();
```

Comment:
can we do (obj == null) ? obj : obj.toString() ?

Reply:
The return type is String for this function. That is why I am returning null over here. If I change it to (obj == null) ? obj : obj.toString(), it throws compile time error since obj is just an object.